### PR TITLE
feat: add logging to pacstall functions

### PR DESCRIPTION
**Pull request by Henryws**
## Purpose

Currently we don't have a way of viewing large compilation logs in Pacstall.

## Approach

Redirect stdout and stderr to a log file, and purge month old files.

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -312,6 +312,13 @@ function generate_changelog() {
     echo -e " -- $maintainer  $(date +"%a, %d %b %Y %T %z")"
 }
 
+function clean_logdir() {
+    local files=("$(find -H "/var/log/pacstall/error_log/" -maxdepth 0 -mtime +30)")
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:316]:
```suggestion
    local files=("$(find -H "/var/log/pacstall/error_log/" -maxdepth 1 -mtime +30)")
```

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -855,7 +862,8 @@ function fail_out_functions() {
 function run_function() {
     local func="$1"
     fancy_message sub "Running $func"
-    $func
+    # https://stackoverflow.com/a/29163890 (shorthand for 2>&1 |)
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:865]:
```suggestion
    # NOTE: https://stackoverflow.com/a/29163890 (shorthand for 2>&1 |)
```

**Review by wizard-28 — CHANGES_REQUESTED** [review]

**Review by wizard-28 — APPROVED** [review]

**Comment by D-Brox** [comment]
Fix the pre-commit check failure

**Comment by Henryws** [comment]
> Fix the pre-commit check failure

The failure is something I already fixed.

**Review by D-Brox — APPROVED** [review]

**Pull request merged**